### ISSUE TITLE
Autosuggest error: "Use a string value as the first field specified for autosuggest!"

### ISSUE THREAD

**Issue by mesosteros** [matteodem/meteor-easy-search#207]
I'm trying to implement autosuggest but it gives me this error. The regular search returns results and I made sure that my first field is a string value.
I'm tring to return values from the Visit Tracker meteor package (https://atmospherejs.com/bigdata/visit-tracker) namely IP addresses, browser names, etc.
I just can't get the suggestion to work. I don't know if it is because the collection is nested (I have to call it like VisitTracker.visits). I'm also using Carbon Router but I don't think this is related (data subscription is returning after all).

In my lib folder's collection.js I added:

```javascript
EasySearch.createSearchIndex('visits', {
    'field' : ['userAgent.browser.family', 'ipAddress', 'createdAt', 'geo', 'returnVisits'],
    'collection' : VisitTracker.visits,
    'limit' : 20,
    'use': 'mongo-db',
    'query' : function (searchString, opts) {
        var query = EasySearch.getSearcher(this.use).defaultQuery(this, searchString);
        return query;
    }
});
```
And in the html:

```html
<template name="altSearch">
    {{> esAutosuggest index="visits" renderSuggestion=suggestion }}
    <ul>
        {{#esEach index="visits"}}
            <li>{{createdAt}}, IP: {{ipAddress}} , userAgentData: {{userAgent.browser.family}}</li>
        {{/esEach}}
    </ul>
    {{> esLoadMoreButton index="visits"}}
    {{#ifEsHasNoResults index="visits"}}
        <div class="no-results">No results found!</div>
    {{/ifEsHasNoResults}}
    {{#ifEsIsSearching index="visits"}}
            <div>Loading...</div>
    {{/ifEsIsSearching}}
</template>
<template name="suggestionTpl">
    {{pre}}<span class="found">{{found}}</span>{{post}}
    {{{doc.userAgent.browser.family}}}
</template>
```

Finally the js:

```javascript
Template.altSearch.onCreated(function () {
  this.autorun(function () {
    var instance = EasySearch.getComponentInstance(
        { index : 'visits' }
    );
    instance.on('autosuggestSelected', function (values) {
      // run every time the autosuggest selection changes
    });
  });
});
Template.altSearch.helpers({
  'suggestion' : function () {
     return Template.suggestionTpl;
  }
});
```

Here is the error:
```
Exception in template helper: Error: Use a string value as the first field specified for autosuggest!
    at Object.EasySearch.getComponentInstance.Template.esAutosuggest.helpers.snippets (http://localhost:3000/packages/matteodem_easy-search.js?19cc9f15488fd08f6c512b6858f4e98a2715cc16:1665:15)
    at bindDataContext (http://localhost:3000/packages/blaze.js?a5c324925e5f6e800a4c618d71caf2848b53bf51:2880:16)
    at Blaze._wrapCatchingExceptions (http://localhost:3000/packages/blaze.js?a5c324925e5f6e800a4c618d71caf2848b53bf51:1651:16)
    at http://localhost:3000/packages/blaze.js?a5c324925e5f6e800a4c618d71caf2848b53bf51:2928:66
    at Function.Template._withTemplateInstanceFunc (http://localhost:3000/packages/blaze.js?a5c324925e5f6e800a4c618d71caf2848b53bf51:3476:12)
    at wrapHelper (http://localhost:3000/packages/blaze.js?a5c324925e5f6e800a4c618d71caf2848b53bf51:2927:27)
    at Spacebars.call (http://localhost:3000/packages/spacebars.js?7bafbe05ec09b6bbb6a3b276537e4995ab298a2f:172:18)
    at Spacebars.mustacheImpl (http://localhost:3000/packages/spacebars.js?7bafbe05ec09b6bbb6a3b276537e4995ab298a2f:109:25)
    at Object.Spacebars.dataMustache (http://localhost:3000/packages/spacebars.js?7bafbe05ec09b6bbb6a3b276537e4995ab298a2f:141:39)
    at http://localhost:3000/packages/matteodem_easy-search.js?19cc9f15488fd08f6c512b6858f4e98a2715cc16:1043:32
```

**Comment by matteodem** [comment]
Please show me an example doc.

**Comment by mesosteros** [comment]
```javascript
{
createdAt: Sat Apr 05 2014 03:42:12 GMT-0700 (PDT),
referer: 'https://www.google.com/search?q=visit+tracker&oq=visit+tracker&aqs=chrome..69i57j69i61l3j0l2.3647j0j8&sourceid=chrome&espv=210&es_sm=119&ie=UTF-8',
ipAddress: '76.167.148.250',
userAgent: {
 raw: 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/33.0.1750.152 Safari/537.36',
 os: {
  family: 'Mac OS X',
  major: '10',
  minor: '9',
  patch: '0',
  patchMinor: null
 },
 device: {
  family: 'Other'
 },
 browser: {
  family: 'Chrome',
  major: 33,
  minor: 0,
  patch: 1750
 }
},
trafficSource: {
 sid: 'googd'
 cmp: '2341234'
 s1: 'ad4342'
 s2: 'dopetracker'
}
geo: {
 range: [ 1286050560, 1286051327 ],
 country: 'US',
 region: 'CA',
 city: 'La Jolla',
 ll: [ 32.8667, -117.2481 ]
}
returnVisits: [
  Sat Apr 05 2014 04:22:05 GMT-0700 (PDT),
  Sat Apr 05 2014 05:45:00 GMT-0700 (PDT)
]
}
```

**Comment by matteodem** [comment]
This has to do with #169

**Comment by matteodem** [comment]
This is solved as of v2, see #169 for a short explanation.
